File: sequences/simple.py
from .basic import naturals, nonnegative_integers


def pentagonal_numbers():
    for i in nonnegative_integers():
        yield n * (3 * n - 1) / 2


def A000326():
    return pentagonal_numbers()

A000326.start = 0


def e_proportion():
    e = 2.718281828
    for i in naturals():
        yield int(i * (1.0 - 1.0 / e))


def e_proportion_remainder():
    for n, i in zip(naturals(), e_proportion()):
        yield n - i

File: sequences/src/Simple.hs
module Simple where

import Basic(numbers)

nth_powers :: Integer -> [Integer]
nth_powers n = map (^ n) numbers

square_numbers :: [Integer]
square_numbers = map square numbers
  where square x = x * x

triangle :: Integer -> Integer
triangle x = x * (x + 1) `div` 2

triangle_numbers :: [Integer]
triangle_numbers = map triangle numbers

pentagon :: Integer -> Integer
pentagon x = x * (3 * x - 1) `div` 2

pentagonal_numbers :: [Integer]
pentagonal_numbers = map pentagon numbers

a000326 :: [Integer]
a000326 = pentagonal_numbers
